pkgs/build-support/cc-wrapper/add-hardening.sh

```diff
@@ -1,32 +1,42 @@
-hardeningFlags=(fortify stackprotector pic strictoverflow format relro bindnow)
-# Intentionally word-split in case 'hardeningEnable' is defined in Nix.
-hardeningFlags+=(${hardeningEnable[@]})
+allHardeningFlags=(fortify stackprotector pie pic strictoverflow format relro bindnow)
 hardeningCFlags=()
 hardeningLDFlags=()
 
-declare -A hardeningDisableMap
+declare -A hardeningDisableMap=()
+declare -A hardeningEnableMap=()
 
-# Intentionally word-split in case 'hardeningDisable' is defined in Nix. The
+# Create table of unsupported flags for this toolchain.
+for flag in @hardening_unsupported_flags@; do
+  hardeningDisableMap[$flag]=1
+done
+
+# Intentionally word-split in case 'NIX_HARDENING_ENABLE' is defined in Nix. The
 # array expansion also prevents undefined variables from causing trouble with
 # `set -u`.
-for flag in ${hardeningDisable[@]} @hardening_unsupported_flags@
-do
-  hardeningDisableMap[$flag]=1
+for flag in ${NIX_HARDENING_ENABLE-}; do
```

Comment:
What is `infixSalt`? I've never seen it before, but all of a sudden Nixpkgs has dozens of references to it...